### DISCUSSION TITLE
Clonar repositório através de scripts do Travis CI

### DISCUSSION THREAD

**Discussion by thiamsantos** [frontendbr/forum#370] [Perguntas e Respostas]
Eu gostaria de Clonar um repositório e fazer o push de alguns arquivos através do travas. 
Mas eu não sei como ligar para clonar. Preciso usar ssh? Tem uma feature do git relacionado a isso? Tem como usar um token do github para isso?

**Comment by bukinoshita** [comment]
# Se o projeto for seu:
1. Cria conta no [TravisCI](https://travis-ci.org)
2. Segue os steps da docs deles

> Enable your projects below by flicking the switch, add a .travis.yml to your project, and push a new commit to GitHub.

# Se o projeto não for seu:

Quando você da um fork em um projeto, ele se torna seu. Só tratar ele do mesmo jeito que você faria com um projeto normal.

## Steps:
1. Fork do projeto
2. Criar uma conta no [TravisCI](https://travis-ci.org)
3. Segue os steps da docs deles
4. Se quiser trocar a flag no **readme**.

> Enable your projects below by flicking the switch, add a .travis.yml to your project, and push a new commit to GitHub.

# Exemplo:

```
// .travis.yml file
language: node_js
node_js:
  - "6"
  - "6.1"
  - "5.11"
  - "0.6"
  - "iojs"
```

**Comment by thiamsantos** [comment]
@BuKinoshita obrigado!!! 

Eu não fui bem claro antes na minha pergunta. Eu gostaria de clonar um repositório diferente, para que eu possar mandar alguns arquivos do repositório atual para esse outro repositório.

**Comment by bukinoshita** [comment]
Se você quer contribuir para um projeto, o ideal é você dar um fork primeiro no projeto, clonar, fazer as mudanças, dar um push e depois abrir um pull request. Quando você der um pull request os testes rodam automaticamente e vai retornar um status se as mudanças passaram no teste ou não, caso tenha passado, o owner do repositório pode dar um merge.

O que você pode fazer também é rodar os testes localmente.

**Comment by thiamsantos** [comment]
@BuKinoshita eu alterei a pergunta porque eu queria saber outra coisa. Mas muito obrigado por ter respondido antes. 

Mas agora me expressando melhor. Eu tenho um repositório A que consiste basicamente em um plugin stylus e algum javascript. O meu objetivo é que após dar um Push o Travis,  clone um repositório B e mande o código stylus do repositório A para esse repositório B e que o código javascript seja mandado para um repositório C. 
Daí entra minha dificuldade, como eu me autêntico dentro do Travis para clonar esse outros repositórios?
